test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,181 @@
+package firewall
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+)
+
+type Firewall struct {
+	ns           string
+	utilsHelpers utils.UtilsInterface
+}
+
+func New(ns string, utilsHelpers utils.UtilsInterface) *Firewall {
+	return &Firewall{
+		ns:           ns,
+		utilsHelpers: utilsHelpers,
+	}
+}
+
+func (firewall *Firewall) runBashCommandOnPod(debugPod *v1.Pod, command string, chrootDir string) ([]byte, error) {
+	var fullCommand string
+	if chrootDir != "" {
+		// Format the command for chroot
+		fullCommand = fmt.Sprintf("chroot %s /bin/bash -c '%s'", chrootDir, command)
+	} else {
+		fullCommand = command
+	}
+
+	output, err := firewall.utilsHelpers.RunCommandOnPod(debugPod, []string{"bash", "-c", fullCommand})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute command '%s' on node %s: %w", command, debugPod.Spec.NodeName, err)
+	}
+	return output, nil
+}
+
+func (firewall *Firewall) ApplyRulesToNode(NFTtable []byte, nodeName string) error {
+	debugPod, err := firewall.utilsHelpers.CreatePodOnNode(nodeName, firewall.ns, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := firewall.utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	_, err = firewall.runBashCommandOnPod(debugPod, fmt.Sprintf("echo '%s' > /host/etc/nftables/firewall.nft", string(NFTtable)), "")
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	_, err = firewall.runBashCommandOnPod(debugPod, "nft -f /etc/nftables/firewall.nft", "/host")
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set on node %s: %w", nodeName, err)
+	}
+
+	output, err := firewall.nftList(debugPod)
```

Comment:
All the firewall.* funcs are internal, why applying them of the struct?

Reply:
It makes the code complicated and less readable instead of just receive ns for example, the code is full of firewall.*